include/pocl_cache.h

```diff
@@ -48,16 +48,6 @@ pocl_cache_create_program_cachedir(cl_program program, unsigned device_i,
 
```

Comment:
Based on your commit description: What happens if thread A manages to build the kernel and start running it while thread B is simultaneously building the same kernel and then renames it, overwriting the .so loaded and being executed by A?

Reply:
Overwriting the .so should be safe, but i just realized we might be loading the same (or two different b/c of rewrite) kernel.so files with dlopen(). I'll add a commit to fix up dlcache handling a bit.